lib/puppet/module/task.rb

```diff
@@ -61,6 +62,58 @@ def self.is_tasks_filename?(path)
       return true
     end
 
+    def self.get_file_details(path, name_start)
+      unless File.absolute_path(path) == File.path(path)
+        msg = _("File pathnames cannot include relative paths")
+        raise InvalidMetadata.new(msg, 'puppet.tasks/invalid-metadata')
+      end
+
+      # This gets the path from the starting point onward
+      # For files this should be the file subpath from the metadata
+      # For directories it should be the directory subpath plus whatever we globbed
+      name = "#{name_start}#{path.partition(name_start).last}"
```

Comment:
I suspect this check is too generic for some real world conditions but we can fix it later

Reply:
Yeah, we could have something like `stdlib/files/stdlib/file.txt`.